phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -465,7 +465,14 @@ private static String toString(List<byte[]> gps) {
     }
 
     private List<List<Scan>> getParallelScans() throws SQLException {
-        return getParallelScans(EMPTY_BYTE_ARRAY, EMPTY_BYTE_ARRAY);
+        if (scan == null
+                || (ScanUtil.isLocalIndex(scan)
+                        && Bytes.compareTo(context.getScan().getStartRow(), scan.getStartRow()) == 0 && Bytes
+                        .compareTo(context.getScan().getStopRow(), scan.getStopRow()) == 0)) {
+            return getParallelScans(EMPTY_BYTE_ARRAY, EMPTY_BYTE_ARRAY);
```

Comment:
This seems strange - why do we call getParallelScans with empty byte arrays in this case? Please document.